tests/test_external_account.py

```diff
@@ -275,9 +275,77 @@ def assert_resource_manager_request_kwargs(
         assert request_kwargs["headers"] == headers
         assert "body" not in request_kwargs
 
+    def test_valid_token_url_shall_pass_validation(self):
+        # valid url doesn't throw exception, a None value should be return
+        assert not external_account.Credentials.validate_token_url(self.TOKEN_URL)
+
+    def test_token_url_pattern_matching(self):
+        # matching *.sts.googleapis.com
```

Comment:
Can you split this into two? One test for valid URLs and one for invalid. 

You can create a list of URLs and loop over them instead of doing them one by one.

See: https://github.com/googleapis/google-auth-library-java/pull/717/files#diff-39915c4590d0657b4e40f8ea980af48fe9cc302a74db01ae5a323d7bc4110f7dR330

Same below.